modules/preview-react/_examples/stories/examples/TextInputWithReactHookForm.tsx

```diff
@@ -13,7 +13,7 @@ type YupValidationResolver = <T extends {}>(
   validationSchema: SchemaOf<T>
 ) => (data: T) => Promise<{values: T; errors: {}} | {values: {}; errors: FieldErrorsImpl<T>}>;
 
-const useYupValidationResolver: YupValidationResolver = validationSchema => {
+const useYupValidationResolver: any = validationSchema => {
```

Comment:
I'm a little surprised to see you resort to `any` but I figure there's a reason.
- [ ] there's a reason

Reply:
This was a while ago, but I upped the TS version to 4.2 to [get more useful type information](https://devblogs.microsoft.com/typescript/announcing-typescript-4-2/#smarter-type-alias-preservation) and TS 4.2 threw an error here. I spent quite some time trying to figure out a way to type this correctly. It was a rabbit hole of exploding complexity and will-bending.

My options were:
- Become an expert on the `yup` library types
- Throw on an `any`.

This story already required us to add [skipLibCheck](https://github.com/Workday/canvas-kit/blob/ada27da72fc5e854e1824f1c50a17e85dfb8c6ac/tsconfig.json#L22), so I'm already unhappy with this story.